### PR TITLE
React - Tests fix

### PR DESCRIPTION
**Pull request by danybony**
This PR restores the tests that were previously commented:
#### oauth-helper.test
For this it was necessary to unmock the dependencies of the module (crypto-js) as well, so that the real ones were used, instead of the automatically-mocked ones.

#### deep-linking-facade.ios-test
For this it has been a long job if finding the right configuration for `packaje.json`.
The current one mocks most of the native components and I explicitly defined the mock of the native component used by the module under test (`Linking`).

The right configuration was found in https://github.com/cliedeman/jest_test

### CODE REVIEW

File: ReactNative/ReactTwitter/__tests__/core/service/deep-linking-facade.ios-test.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-jest.dontMock('../../../core/service/deep-linking-facade.ios.js')
+jest.unmock('../../../core/service/deep-linking-facade.ios.js')
```
Comment (danybony) [inline ReactNative/ReactTwitter/__tests__/core/service/deep-linking-facade.ios-test.js]: `dontMock` is deprecated now

File: ReactNative/ReactTwitter/__tests__/core/service/oauth-helper.test.js [hunk 2]
```diff
@@ -1,63 +1,64 @@
 jest.unmock('../../../core/service/oauth-helper.js')
 jest.unmock('../../../core/service/percent-encoder.js')
-// var OauthHelper = require('../../../core/service/oauth-helper.js')
+jest.unmock('crypto-js')
+jest.unmock('crypto-js/hmac-sha1')
```
Comment (danybony) [inline ReactNative/ReactTwitter/__tests__/core/service/oauth-helper.test.js]: These two lines are the only change of the file, except for the un-commented code

File: ReactNative/ReactTwitter/jestSupport/Linking.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+/* global jest */
+
+var Linking = {
+  addEventListener: jest.genMockFunction(),
+  removeEventListener: jest.genMockFunction()
+}
+
+module.exports = Linking
```
Comment (danybony) [inline ReactNative/ReactTwitter/jestSupport/Linking.js]: This is our mock of the native component we need for the test

File: ReactNative/ReactTwitter/jestSupport/env.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+jest.setMock('ErrorUtils', {
+  setGlobalHandler () {
+    // TODO
+  }
+})
+
+import NativeModules from './NativeModules'
+import Linking from './Linking'
+jest.setMock('NativeModules', NativeModules)
+jest.setMock('Linking', Linking)
```
Comment (danybony) [inline ReactNative/ReactTwitter/jestSupport/env.js]: Here we can set more mocked components in case we need it for other tests

**Comment by gbasile** [comment]
LGTM 👍

**Comment by eduardb** [comment]
👏

File: ReactNative/ReactTwitter/__tests__/core/service/oauth-helper.test.js [hunk 5]
```diff
@@ -1,63 +1,64 @@
...
-  // })
+  describe('URL Callback', () => {
+    it('should be defined', () => {
+      let urlCallback = 'react-twitter-oauth://callback?oauth_token=xOL-GwAAAAAAvI9oAAABVKmtBm8&oauth_verifier=pFIFpsIVlT59NpDkID3oa3X67UEJLa4k'
+      let result = OauthHelper.getOauthTokenAndVerifierFromURLCallback(urlCallback)
+
+      expect(result).toBeDefined()
+    })
+
+    it('should contains the correct oauth_token', () => {
```
Comment (frapontillo) [inline ReactNative/ReactTwitter/__tests__/core/service/oauth-helper.test.js]: le grammar

File: ReactNative/ReactTwitter/__tests__/core/service/oauth-helper.test.js [hunk 6]
```diff
@@ -1,63 +1,64 @@
...
+    it('should contains the correct verifier', () => {
+      let urlCallback = 'react-twitter-oauth://callback?oauth_token=xOL-GwAAAAAAvI9oAAABVKmtBm8&oauth_verifier=pFIFpsIVlT59NpDkID3oa3X67UEJLa4k'
+      let result = OauthHelper.getOauthTokenAndVerifierFromURLCallback(urlCallback)
+
+      expect(result.oauth_verifier).toBe('pFIFpsIVlT59NpDkID3oa3X67UEJLa4k')
+    })
+  })
+
+  describe('_collectParameters', () => {
+    it('when collecting parameters', () => {
```
Comment (frapontillo) [inline ReactNative/ReactTwitter/__tests__/core/service/oauth-helper.test.js]: should and stuff

File: ReactNative/ReactTwitter/jestSupport/Linking.js (hunk 3 above)
Comment (frapontillo) [inline ReactNative/ReactTwitter/jestSupport/Linking.js]: If this is a mock you should append something like `.mock` or `-mock` in the end of the file name.

File: ReactNative/ReactTwitter/jestSupport/NativeModules.js [hunk 7]
```diff
@@ -0,0 +1,62 @@
+/* global jest */
+/*
+ Taken from react-native __mocks__
+ */
+var NativeModules = {
```
Comment (frapontillo) [inline ReactNative/ReactTwitter/jestSupport/NativeModules.js]: See previous comment for mock filenames.

Comment (frapontillo) [inline ReactNative/ReactTwitter/jestSupport/NativeModules.js]: Good mocking though. 👍

**Comment by frapontillo** [comment]
@gbasile could steal the flow configuration from https://github.com/cliedeman/jest_test as well.

**Comment by danybony** [comment]
@frapontillo comments addressed

**Comment by frapontillo** [comment]
🐑 it

**Pull request merged**